File: Shared/Dto/Underlying/ForWebApi/TraceDto.cs
using System;

namespace Iit.Fibertest.Dto
{
    public class TraceDto
    {
        public Guid TraceId;
        public Guid RtuId;
        public string Title;
        public OtauPortDto OtauPort;
        public bool IsAttached;
        public int Port;

        public FiberState State;

        public bool HasEnoughBaseRefsToPerformMonitoring;
        public bool IsIncludedInMonitoringCycle;
    }
}

File: AngularClientApp/src/app/models/dtos/traceDto.ts
import { OtauDto } from './otauDto';
import { FiberState } from '../enums/fiberState';

export interface TraceDto {
  traceId: string;
  rtuId: string;
  title: string;

  otauPort: OtauDto;
  isAttached: boolean;
  port: number;

  state: FiberState;

  hasEnoughBaseRefsToPerformMonitoring: boolean;
  isIncludedInMonitoringCycle: boolean;
}
